lib/debug/client.rb

```diff
@@ -56,6 +76,15 @@ def cleanup_unix_domain_sockets
       def list_connections
         Dir.glob(DEBUGGER__.create_unix_domain_socket_name_prefix + '*')
       end
+
+      def current_shell
+        case `ps -p #{Process.ppid} -o 'args='`.strip
+        when 'fish', '-fish'
+          :fish
```

Comment:
I'm not sure whether ps works on Windows or not. I googled it to see (I don't have a Windows box to try it on), and the internet suggested that it's called tasklist on Windows.

Anyway, when I run it, it returns the full path, which won't match the case statement:

```sh
$ ruby -e 'p `ps -p #{Process.ppid} -o args=`.strip'
"/opt/homebrew/bin/fish"
```

Probably a regex would be more resilient, something like `/\bfish\z/`

Reply:
Thanks, didn't think about Windows, I can't test on Windows though

Reply:
I did `File.basename(...)` so that it works on your setup too, on mine I get:

```
> ruby -e 'p `ps -p #{Process.ppid} -o args=`.strip'
"-fish"
```

Reply:
I'm downloading a Windows VM to test on Windows

Reply:
On windows, there is no place to use this feature, so no problem.

Reply:
yeah adding windows support would be another larger pull request